protostar-rust/src/lib.rs

```diff
@@ -134,6 +184,66 @@ fn run_tests(tests: TestsFromFile, tests_stats: &mut TestsStats) -> Result<()> {
     Ok(())
 }
 
+fn strip_path_from_test_names(test_configs: Vec<TestConfig>) -> Result<Vec<TestConfig>> {
+    let mut result = vec![];
+    for test_config in test_configs {
+        let name: String = test_config
+            .name
+            .clone()
+            .rsplit('/')
+            .next()
+            .with_context(|| format!("Failed to get test name from = {}", test_config.name))?
+            .into();
+        result.push(TestConfig {
+            name,
+            available_gas: test_config.available_gas,
+        });
+    }
+    Ok(result)
+}
+
+fn filter_tests_by_name(
+    test_name_filter: &str,
+    exact_match: bool,
+    test_configs: Vec<TestConfig>,
+) -> Result<Vec<TestConfig>> {
+    let mut result = vec![];
+    for test in test_configs {
+        if exact_match {
+            if test.name == test_name_filter {
+                result.push(test);
+            }
+        } else if test_name_contains(test_name_filter, &test)? {
+            result.push(test);
+        }
+    }
+    Ok(result)
+}
+
+fn test_name_contains(test_name_filter: &str, test: &TestConfig) -> Result<bool> {
+    let name = test
+        .name
+        .rsplit("::")
+        .next()
+        .context(format!("Failed to get test name from = {}", test.name))?;
+    Ok(name.contains(test_name_filter))
+}
+
+pub fn protostar_config_for_package(
+    metadata: &Metadata,
+    package: &PackageId,
+) -> Result<ProtostarConfigFromScarb> {
+    let raw_metadata = metadata
+        .get_package(package)
+        .ok_or_else(|| anyhow!("Failed to find metadata for package = {package}"))?
+        .tool_metadata("protostar")
+        .ok_or_else(|| anyhow!("Failed to find protostar config for package = {package}"))?
+        .clone();
+    let protostar_config: ProtostarConfigFromScarb = serde_json::from_value(raw_metadata)?;
+
+    Ok(protostar_config)
+}
+
```

Comment:
I think we can make the config not required in this PR already. Since we removed it last time, it wouldn'tt increase the scope of the PR